CUJExecutor/app/src/androidTest/java/com/example/firstapp/ShellUtilityTest.java

```diff
@@ -331,11 +362,108 @@ public void properSubstringAndClickingOnNonClickables() throws Exception {
 
     @Test
     public void clickTestCached() throws Exception {
-        shellUtility.launchApp(BASIC_SAMPLE_PACKAGE);
         String preCUJ = "";
-        String postCUJ = "['start;" + BASIC_SAMPLE_PACKAGE + "', 'click;COUNT', 'click;COUNT', 'click;COUNT', 'click;COUNT', 'click;COUNT']";
-        shellUtility.iterateAndMeasureCuj(preCUJ, postCUJ, 1,false, 0);
-        UiObject2 five = shellUtility.device.wait(Until.findObject(By.textContains("5")), TIMEOUT);
+        String measuredCUJ = "['start;" + BASIC_SAMPLE_PACKAGE + "', 'click;COUNT', 'click;COUNT', 'click;COUNT', 'click;COUNT']";
+        String postCUJ = "['click;COUNT']";
+
+        shellUtility.iterateAndMeasureCuj(preCUJ, measuredCUJ, postCUJ, 1,false, 0);
+        UiObject2 five = shellUtility.device.wait(Until.findObject(By.text("5")), TIMEOUT);
         assertNotEquals(null, five);
     }
-}
\ No newline at end of file
+
+    /**
+     * Walk CUJ Tests
+     */
+    @Test
+    public void walkCujW() throws Exception {
+        String preCUJ = "['start;" + BASIC_SAMPLE_PACKAGE + "']";
+        String measuredCUJ = "['click;RANDOM']";
+        String postCUJ = "['click;PREVIOUS', 'click;COUNT']";
+        int n = 2;
+
+        String sectionCUJ = "w";
+        shellUtility.parseUserWalkCujNTimes(preCUJ, measuredCUJ, postCUJ, sectionCUJ, n);
```

Comment:
Needs a better name.  `parseUserWalkCujNTimes()` implies that it is returning something.